pkg/controllers/ttl-controller/manager.go

```diff
@@ -0,0 +1,186 @@
+package ttlcontroller
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/kyverno/kyverno/pkg/auth/checker"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/metadata"
+	"k8s.io/client-go/metadata/metadatainformer"
+	"k8s.io/client-go/tools/cache"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	authorizationv1client "k8s.io/client-go/kubernetes/typed/authorization/v1"
+)
+
+type stopFunc = context.CancelFunc
+
+type manager struct {
+	metadataClient  metadata.Interface
+	discoveryClient discovery.DiscoveryInterface
+	checker         checker.AuthChecker
+	resController   map[schema.GroupVersionResource]stopFunc
+}
+
+func NewManager(metadataInterface metadata.Interface, discoveryInterface discovery.DiscoveryInterface,  authorizationInterface authorizationv1client.AuthorizationV1Interface) (*manager, error) {
+	// client
+	metadataClient := metadataInterface
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// discovery client
+	discoveryClient:= discoveryInterface
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	authClient := authorizationInterface
+
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// checker
+
+	selfChecker := checker.NewSelfChecker(authClient.SelfSubjectAccessReviews())
+
+	resController := make(map[schema.GroupVersionResource]stopFunc)
+
+	return &manager{
```

Comment:
You may want to embed the logger and waitgroup to the manager just as other controllers do.